crates/tracing-instrumentation/src/lib.rs

```diff
@@ -51,23 +50,86 @@ pub enum Error {
     LogDirectiveParseError(#[from] ParseError),
 }
 
+/// creates and register a global opentelemetry tracer provider. The global
+/// provider is mainly used by the [`invocation_span!`] macro
+fn build_services_tracing(common_opts: &CommonOptions) -> Result<(), Error> {
+    let opts = &common_opts.tracing;
+
+    let endpoint = match &opts
+        .tracing_services_endpoint
+        .as_ref()
+        .or(opts.tracing_endpoint.as_ref())
+    {
+        Some(endpoint) => *endpoint,
+        None => return Ok(()),
+    };
+
+    let resource = opentelemetry_sdk::Resource::new(vec![
+        KeyValue::new(
+            opentelemetry_semantic_conventions::resource::SERVICE_NAME,
+            "services",
+        ),
+        KeyValue::new(
+            opentelemetry_semantic_conventions::resource::SERVICE_NAMESPACE,
+            "Restate",
+        ),
+        KeyValue::new(
+            opentelemetry_semantic_conventions::resource::SERVICE_INSTANCE_ID,
+            format!("{}/{}", common_opts.cluster_name(), common_opts.node_name()),
+        ),
+        KeyValue::new(
+            opentelemetry_semantic_conventions::resource::SERVICE_VERSION,
+            env!("CARGO_PKG_VERSION"),
+        ),
+    ]);
+
+    let header_map = HeaderMap::from_iter(HashMap::from(opts.tracing_headers.clone()));
+
+    let exporter = SpanExporterBuilder::from(
+        opentelemetry_otlp::new_exporter()
+            .tonic()
+            .with_endpoint(endpoint)
+            .with_metadata(MetadataMap::from_headers(header_map.clone())),
+    )
+    .build_span_exporter()?;
+
+    let exporter = ResourceModifyingSpanExporter::new(exporter);
+
+    let provider = opentelemetry_sdk::trace::TracerProvider::builder()
+        .with_config(opentelemetry_sdk::trace::Config::default().with_resource(resource))
+        .with_span_processor(
+            BatchSpanProcessor::builder(exporter, opentelemetry_sdk::runtime::Tokio).build(),
+        )
+        .build();
```

Comment:
I actually checked this and I couldn't find away to do filtering from open-telemetry sdk. Even the level info we provide  to the micro is not actually used at all with the open-telemetry spans but I left it in case this changes in the future by a custom implementation.

I might be wrong, but open telemetry has no notion of Level, or Filter. The only possible span information associated with the span can be seen here [SpanBuilder](https://docs.rs/opentelemetry/0.24.0/opentelemetry/trace/struct.SpanBuilder.html).

What we can do is to maybe build our own TracerProvider that understands some sort of filtering and then decide to build the span or not. The macro can try to do an early check if the span should be emitted against the tracer before it continues.

For the **sampling**, it's fully supported of course by optl and can be configured. Again I think this can be part of a future PR but it's possible. it's the filtering that probably require more work